crates/holochain_cascade/src/lib.rs

```diff
@@ -1021,37 +1019,40 @@ impl CascadeImpl {
         }
     }
 
+    /// Get agent activity from agent activity authorities.
+    ///
+    /// Hashes are requested from the authority and cache for valid chains.
+    ///
+    /// Query:
+    /// - [include_entries](ChainQueryFilter::include_entries) will also fetch the entries in parallel (requires include_full_actions)
+    /// - [sequence_range](ChainQueryFilter::sequence_range) will get all the activity in the exclusive range
+    /// - [action_type](ChainQueryFilter::action_type) and [entry_type](ChainQueryFilter::entry_type) will filter the activity (requires include_full_actions)
+    ///
+    /// Options:
+    /// - [include_valid_activity](GetActivityOptions::include_valid_activity) will include the valid chain hashes.
+    /// - [include_rejected_activity](GetActivityOptions::include_rejected_activity) will include the invalid chain hashes.
+    /// - [include_warrants](GetActivityOptions::include_warrants) will include the warrants for this agent.
+    /// - [include_full_records](GetActivityOptions::include_full_records) will also fetch the entries in parallel (requires include_full_actions)
+    /// - [include_full_actions](GetActivityOptions::include_full_actions) will fetch the valid actions in parallel (requires include_valid_activity)
```

Comment:
Do these `(require *)` auto enable that other item? Or does it error if you don't also enable that?

Reply:
These docs are wrong, and that reminds me that I've missed something in the top level call. I'm not respecting the query for whether entries should be retrieved. This needs more tests really! I'll correct the docs and fix the oversight but I want to go over this code again later

Reply:
and to answer the question, no. The options don't enable each other and you have to set what you mean. I think we should expose a simpler interface with fewer options to the HDK when we come to update it. I've tried to avoid breaking changes for now so that this can be tested easily with the current version of HF